server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/cache/CassandraDumbBlobStoreCache.java

```diff
@@ -0,0 +1,165 @@
+import static com.datastax.driver.core.querybuilder.QueryBuilder.delete;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.eq;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.insertInto;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.select;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.ttl;
+import static org.apache.james.blob.cassandra.BlobTables.BucketBlobTable.ID;
+import static org.apache.james.blob.cassandra.BlobTables.DumbBlobCache.DATA;
+import static org.apache.james.blob.cassandra.BlobTables.DumbBlobCache.TABLE_NAME;
+import static org.apache.james.blob.cassandra.BlobTables.DumbBlobCache.TTL;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+import javax.inject.Inject;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.james.backends.cassandra.utils.CassandraAsyncExecutor;
+import org.apache.james.blob.api.BlobId;
+import org.reactivestreams.Publisher;
+
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.Session;
+import com.google.common.base.Preconditions;
+
+import reactor.core.publisher.Mono;
+
+public class CassandraDumbBlobStoreCache implements DumbBlobStoreCache {
+
+    private final CassandraAsyncExecutor cassandraAsyncExecutor;
+    private final PreparedStatement insertStatement;
+    private final PreparedStatement selectStatement;
+    private final PreparedStatement deleteStatement;
+    private final CassandraCacheConfiguration cacheConfiguration;
+
+    @Inject
+    public CassandraDumbBlobStoreCache(Session session, CassandraCacheConfiguration cacheConfiguration) {
+        this.cassandraAsyncExecutor = new CassandraAsyncExecutor(session);
+        this.insertStatement = prepareInsert(session);
+        this.selectStatement = prepareSelect(session);
+        this.deleteStatement = prepareDelete(session);
+        this.cacheConfiguration = cacheConfiguration;
+    }
+
+    @Override
+    public Publisher<Void> cache(BlobId blobId, byte[] bytes) {
+        Preconditions.checkNotNull(bytes, "Data must not be null");
+        Preconditions.checkArgument(bytes.length > 0, "Data must not be empty");
+
+        return Mono.just(bytes)
+            .flatMap(data -> {
+                if (data.length > cacheConfiguration.getByteThresholdSize()) {
+                    return Mono.empty();
+                } else {
+                    return Mono.from(save(blobId, ByteBuffer.wrap(data, 0, data.length)));
+                }
+            });
+    }
+
+    private Publisher<Void> save(BlobId blobId, ByteBuffer data) {
+        return cassandraAsyncExecutor.executeVoid(
+            insertStatement.bind()
+                .setString(ID, blobId.asString())
+                .setBytes(DATA, data)
+                .setInt(TTL, Long.valueOf(cacheConfiguration.getTtl().getSeconds()).intValue())
```

Comment:
https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#toIntExact-long- is your friend, beware of the overflow.